[WIP] Bootstrap 5 fixes AND new sample list view

jbflo opened the pull request:
In this PR 
- We fix bootstrap 5 and css error
- new components for viewing samples list as a table filtered by cell and puck 
- we added possibility to have 3 view mode for the sample list

jbflo committed:
rebase of "jb-bootstrap5"

jbflo committed:
Update validation schema

jbflo committed:
Equipment view, the status bar should cover the entire panel header

Equipment view, no margin on child panels

marcus-oscarsson committed:
Added the possibility to handle a generic task type

marcus-oscarsson committed:
Started to fix tests

marcus-oscarsson committed:
SSX Features

marcus-oscarsson committed:
Temporary MJPEG test

jbflo committed:
Add Cell and puck number to sample data

jbflo committed:
New Sample Grid 

redesigned sample grid view component, adding new Filter Options

jbflo committed:
New Sample Grid 

redesigned sample grid view component, adding new Filter Options

jbflo committed:
Temporary add New Sample Grid Tab

jbflo committed:
Temporary add New Sample Grid Tab

jbflo committed:
UI Update for New Sample Grid view

jbflo committed:
UI Update for New Sample Grid view

jbflo committed:
adding possibility to change sample grid  View Mode 

Flex Grid view on going process

jbflo committed:
adding possibility to change sample grid  View Mode 

Flex Grid view on going process

jbflo committed:
Remove on broadcast eroor due to createStateSyncMiddleware

by blacklisted some redux-persit

jbflo committed:
remove unnecessary key property

jbflo committed:
add  react-copy-to-clipboard  package

marcus-oscarsson committed:
temp

marcus-oscarsson committed:
temp

marcus-oscarsson committed:
Fixed identation

marcus-oscarsson committed:
Fixed identation

Commit:
Temporarily disabling autologout

Commit:
Temporarily disabling autologout

marcus-oscarsson committed:
Added signout on browser exit

marcus-oscarsson committed:
Added signout on browser exit

jbflo committed:
New Said Update

jbflo committed:
New Said Update

jbflo committed:
Sample grid couple with Flex View Mode

jbflo committed:
Sample grid couple with Flex View Mode

jbflo committed:
Fix Context Menu Position

marcus-oscarsson committed:
Using "global role" instead of "generated" id

marcus-oscarsson committed:
Using "global role" instead of "generated" id

Commit:
Removed unused code

Commit:
Removed unused code

Commit:
Fixed default parameter passing for sample grid

Commit:
Adapted to AbstractMachineInfo interface

Commit:
Changed incorrect reference to HWR

Commit:
Removed PDB committed by mistake

Commit:
Removed PDB committed by mistake

Commit:
not passing boolean

Commit:
Passing defualt parameters

Commit:
Passing defualt parameters

Commit:
Replaced use of hook with function

Commit:
Replaced use of hook with function

Commit:
Fix state change color , badge use bg instead of variant

Commit:
Fix state change color , badge use bg instead of variant

Commit:
Card view Styling

Commit:
Card view Styling

Commit:
styling for copy sample name icon
and for selected sample icon

Commit:
styling for copy sample name icon
and for selected sample icon

Commit:
add id to Form check

Commit:
replace map to for Each

Commit:
replace map to for Each

jbflo committed:
Remove unused function and update context menu and grid grid table style

jbflo committed:
Remove unused function and update context menu and grid grid table style

Commit:
Fix menu disappear on scroll

Commit:
Fix menu disappear on scroll

jbflo committed:
remove unnecessary variable

jbflo committed:
remove unnecessary variable

jbflo committed:
Context menu Update

jbflo committed:
Context menu Update

Commit:
Fixed Collect Dialog Position overflow when sample too long

Commit:
Fixed Collect Dialog Position overflow when sample too long

marcus-oscarsson committed:
Fixed broken tests

jbflo committed:
Temporary add new sample to Cell 1 , Puck 1

jbflo committed:
Adding new build in Context Menu component 

Replace react-contecMenu to a Generic Context Menu

jbflo committed:
Adding new build in Context Menu component 

Replace react-contecMenu to a Generic Context Menu

jbflo committed:
Update Flex Pie View 

If Cell is empty disabled pointer event

jbflo committed:
Update Flex Pie View 

If Cell is empty disabled pointer event

Commit:
Fix  Expected an assignment or function call

Commit:
Fix  Expected an assignment or function call

Commit:
Adding puck code/ id to view

Commit:
Adding puck code/ id to view

jbflo committed:
FIx context menu animation

jbflo committed:
Fix Modal position

jbflo committed:
Fix Modal position

jbflo committed:
Fix Modal position

jbflo committed:
Fix Modal position

jbflo committed:
add menu Header

jbflo committed:
add menu Header

jbflo committed:
Fix menu position when  off-screen

jbflo committed:
Change cursor on Menu

jbflo committed:
Change cursor on Menu

jbflo committed:
Remove  unused and fix  css tag and

jbflo committed:
Remove  unused and fix  css tag and

jbflo committed:
Fix dialog position for all

jbflo committed:
Fix dialog position for all

jbflo committed:
remove unused properties and line of code

jbflo committed:
remove unused properties and line of code

jbflo committed:
Make old sample visible as  third view mode on Sample List

remove duplicated code

jbflo committed:
Make old sample visible as  third view mode on Sample List

remove duplicated code

jbflo committed:
Rename Files

make different from old sample grid view to new sample grid table

jbflo committed:
Rename Files

make different from old sample grid view to new sample grid table

jbflo committed:
Remove css style on Table

jbflo committed:
Context Menu repositioning when off the screen

jbflo committed:
Rename to SampleListViewContainer

jbflo committed:
add nowrap style

jbflo committed:
Fix button position

jbflo committed:
Added log level command line switch

jbflo committed:
Fix web socket  proxy problem

jbflo committed:
Remove empty tag css

meguiraun commented:
hi! Looks a great and interesting amount of work!

I was trying to run it but I have some errors and server will not start, which branch/commit in mxcubecore are you running with?

for example:
in `mxcube3/core/components/queue.py` I had to change
```
from mxcubecore.HardwareObjects.base_queue_entry import QUEUE_ENTRY_STATUS
```
in `mxcube3/core/components/sampleview.py` I had to change
````
from mxcubecore.HardwareObjects.queue_entry import CENTRING_METHOD
````

Now I am getting 
````
Traceback (most recent call last):
  File "/Users/mikegu/Documents/MXCUBE/mxcube3_upstream/mxcube3/app.py", line 99, in init
    MXCUBECore.adapt_hardware_objects(app)
  File "/Users/mikegu/Documents/MXCUBE/mxcube3_upstream/mxcube3/app.py", line 153, in adapt_hardware_objects
    _id = MXCUBECore._get_adapter_id(ho)
  File "/Users/mikegu/Documents/MXCUBE/mxcube3_upstream/mxcube3/app.py", line 114, in _get_adapter_id
    _id = ho.global_role
  File "/Users/mikegu/Documents/MXCUBE/mxcubecore/mxcubecore/BaseHardwareObjects.py", line 798, in __getattr__
    raise AttributeError(attr)
AttributeError: global_role
````

marcus-oscarsson commented:
@meguiraun, yes there are still some things to be sorted out with this PR because it needs some additions to mxcubecore that are not yet merged :) We are preparing a PR for mxcubecore, in the mean time @jbflo marked this PR as WIP so that you can have an idea of what it is. It wont, unfortunately, run for the moment.

meguiraun commented:
Ooooo :( ... It really looks promising...

I will wait then! Thanks for the fast reply!

marcus-oscarsson commented:
:), we hope so too, We can roughly go trough the changes during our next meeting

jbflo committed:
Fix sample name too long on spring grid view

jbflo committed:
Fix missing  parameter

jbflo committed:
Add missing state spread operator

jbflo committed:
remove duplicate prop

jbflo committed:
Add localstorage for view-mode 

persist view-mode property  when user reload page

meghdadyazdi commented on ui/src/components/SampleView/ApertureInput.js:
```diff
@@ -25,7 +25,6 @@ export default class ApertureInput extends React.Component {
           className={inputCSS}
           style={{ float: 'none' }}
-          id="puckFilter"
           defaultValue={this.props.aperture}
```
It seems it should be `value={this.props.aperture}` instead of `defaultValue={this.props.aperture}`. Otherwise it doesn’t get updated in ui when aperture is changed from diffractometer.

jbflo replied:
Correct, thanks for having a look !

jbflo replied:
@meghdadyazdi Have you manage to run this PR yet ?

meghdadyazdi replied:
Not yet actually. As I was working on aperture hardwareObject I noticed this issue.

jbflo committed:
add value instead of defaultValue to select form

jbflo committed:
Give a Unique Key to component !

jbflo committed:
update machine info width and row style

marcus-oscarsson committed:
ServerStorage not working with latest version of ReduxPersist

marcus-oscarsson committed:
Introduced handling of generic task forms, a SSX CHIP control and renamed beamline-attributes to hardware_objects

marcus-oscarsson committed:
Fixed tests

marcus-oscarsson committed:
Adapted to latest mxcubecore changes

- Updated imports for queue_entry and model
- Added logging command line switch
- Using global role to uniquely identify a hardware object within HWR

jbflo committed:
fix Form style and add required prob to DropdownButton

jbflo committed:
Add missing  beamline level prop

jbflo committed:
Fix validation props

jbflo committed:
Remove debugger line

jbflo committed:
Fixed missing prop on  Form Validation

marcus-oscarsson committed:
Small fix for video streaming

  - Removed code committed by accident
  - Linting
  - Added possibility to configure video url

jbflo committed:
check if sample changer has  cell and Basket attribute

jbflo committed:
remove unused Code

jbflo committed:
Add possibility to change view automatically 

Sample Changer Mockup class does not have Cell Level yet, so we keep spring grid view  to display Sample

jbflo committed:
remove non-boolean attribute

jbflo committed:
Remove ESRF specific config

jbflo committed:
Remove View mode Options when You only have a single  Option

When Component does not have Cell level , we remove Filter by Cell Options

jbflo committed:
Check should be in componentDidMount

jbflo commented:
we close this Pr.
And opened a new one
